pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -990,3 +979,135 @@ func TestHighProirotyFlushUnschedulableQLeftover(t *testing.T) {
 		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPod.Name, p.Name)
 	}
 }
+
+// TestHighProirotyPodInfo tests the operations related to podInfo.
+func TestHighProirotyPodInfo(t *testing.T) {
+	pod1 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-1",
+			Namespace: "ns1",
+			UID:       types.UID("tp-1"),
+		},
+		Status: v1.PodStatus{
+			NominatedNodeName: "node1",
+		},
+	}
+
+	pod2 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-2",
+			Namespace: "ns2",
+			UID:       types.UID("tp-2"),
+		},
+		Status: v1.PodStatus{
+			NominatedNodeName: "node2",
+		},
+	}
+
+	pInfo1 := &podInfo{
+		pod:       pod1,
+		timestamp: util.RealClock{}.Now(),
+	}
+	pInfo2 := &podInfo{
+		pod:       pod2,
+		timestamp: util.RealClock{}.Now().Add(1 * time.Second),
+	}
+
+	var queue *PriorityQueue
+	type operation = func()
+	addPodActiveQ := func(pInfo *podInfo) operation {
+		return func() {
+			queue.activeQ.Add(pInfo)
+		}
+	}
+	updatePodActiveQ := func(pInfo *podInfo) operation {
+		return func() {
+			queue.activeQ.Update(pInfo)
+		}
+	}
+	addPodUnschedulableQ := func(pInfo *podInfo) operation {
+		return func() {
+			// Update pod condition to unschedulable.
+			podutil.UpdatePodCondition(&pInfo.pod.Status, &v1.PodCondition{
+				Type:    v1.PodScheduled,
+				Status:  v1.ConditionFalse,
+				Reason:  v1.PodReasonUnschedulable,
+				Message: "fake scheduling failure",
+			})
+			queue.unschedulableQ.addOrUpdate(pInfo)
+		}
+	}
+	addPodBackoffQ := func(pInfo *podInfo) operation {
+		return func() {
+			queue.podBackoffQ.Add(pInfo)
+		}
+	}
+	moveAllToActiveQ := func() operation {
+		return func() {
+			queue.MoveAllToActiveQueue()
+		}
+	}
+	flushBackoffQ := func() operation {
+		return func() {
+			queue.flushBackoffQCompleted()
+		}
+	}
+	tests := []struct {
+		name       string
+		operations []operation
+		expected   []*podInfo
+	}{
+		{
+			name: "add two pod to activeQ and pod them by the timestamp",
+			operations: []operation{
+				addPodActiveQ(pInfo2), addPodActiveQ(pInfo1),
+			},
+			expected: []*podInfo{pInfo1, pInfo2},
+		},
+		{
+			name: "update two pod to activeQ and pod them by the timestamp",
+			operations: []operation{
+				updatePodActiveQ(pInfo2), updatePodActiveQ(pInfo1),
+			},
+			expected: []*podInfo{pInfo1, pInfo2},
+		},
+		{
+			name: "add two pod to unschedulableQ then move them to activeQ and pod them by the timestamp",
+			operations: []operation{
+				addPodUnschedulableQ(pInfo2), addPodUnschedulableQ(pInfo1), moveAllToActiveQ(),
+			},
+			expected: []*podInfo{pInfo1, pInfo2},
+		},
+		{
+			name: "add one pod to BackoffQ and move it to activeQ",
+			operations: []operation{
+				addPodBackoffQ(pInfo1), flushBackoffQ(), moveAllToActiveQ(),
```

Comment:
Could you add `pInfo2` to the activeQ in this test case and check that `pInfo2` is after `pInfo1` in the expected output?

Reply:
Done thank you